buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -369,21 +369,23 @@ class BuildPlugin implements Plugin<Project> {
         project.afterEvaluate {
             // fail on all javac warnings
             project.tasks.withType(JavaCompile) {
-                options.fork = true
-                options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
-                options.forkOptions.memoryMaximumSize = "1g"
+                if (targetCompatibility == '1.8' || targetCompatibility == '1.9') {
+                    options.fork = true
+                    options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
```

Comment:
I assume we will be using java 8 to compile java 7? In that case, i don't see why we need to conditionalize any of the fork options. It seems like it should only be the compact profile arg that is conditionalized below?